### PR TITLE
Re-land "Make `fml/...` compatible with `.clang_tidy`

### PR DESCRIPTION
**Pull request by matanlurey**
Reverts flutter/engine#48004.
Closes https://github.com/flutter/flutter/issues/138381.

### CODE REVIEW

**Comment by matanlurey** [comment]
I have to manually revert the CI YAML change before merging.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
test-exempt: helps unblock turning on a check that will automatically enforce the change

**Pull request merged**
